x-pack/plugins/security/server/lib/authentication/providers/saml.ts

```diff
@@ -476,7 +463,7 @@ export class SAMLAuthenticationProvider {
       // handshake. Obviously we can't do that for AJAX requests, so we just reply with `400` and clear error message.
       // There are two reasons for `400` and not `401`: Elasticsearch search responds with `400` so it seems logical
       // to do the same on Kibana side and `401` would force user to logout and do full SLO if it's supported.
-      if (isInvalidRefreshTokenError(err)) {
+      if (getErrorStatusCode(err) === 400) {
```

Comment:
Is it possible that when we do the access token logic, we get back the 401 because the token document exists and it's expired, and then before we get here the refresh token is deleted and we get back a 500 with the message body denoting the token being deleted?

Also, Can we log the original error message here, just in-case we need to debug a weird 400 being returned in the future triggering this condition?

Reply:
> Is it possible that when we do the access token logic, we get back the 401 because the token document exists and it's expired, and then before we get here the refresh token is deleted and we get back a 500 with the message body denoting the token being deleted?

I believe that shouldn't be possible, see https://github.com/elastic/elasticsearch/blob/9beb31fd3c5a8323cb08cc524f1a2268e9c72c24/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java#L743-L763. There are a couple of cases like missing document or even index, and in all of them we'll get `invalidGrantException("could not refresh the requested token")` that is [`400 Bad Request`](https://github.com/elastic/elasticsearch/blob/9beb31fd3c5a8323cb08cc524f1a2268e9c72c24/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java#L1401-L1406).

> Also, Can we log the original error message here, just in-case we need to debug a weird 400 being returned in the future triggering this condition?

Definitely!

Reply:
Ah, we log this already, see few lines above :) 
```ts
this.options.log(
  ['debug', 'security', 'saml'],
  `Failed to refresh access token: ${err.message}`
);
```

Reply:
I completely missed that we did so above. Please ignore!